drafts/concurrency_demystified.md

```diff
@@ -0,0 +1,109 @@
+> 
+> "Got it," John said, feeling a bit more confident. "But what about async programming?"
+> 
+> "Async programming is a bit different," Dr. Carter explained. "It's like you doing the laundry, but while the washing machine is running, you start cooking. You don't just sit and wait for the laundry to finish. Instead, you switch to another task that doesn't need you to be present the whole time."
+> 
+> John's eyes lit up. "So, parallel is about multiple workers at the same time, and async is about efficient multitasking during idle periods. Got it!"
+> 
+> Dr. Carter grinned. "Exactly. And here's a twist: When you combine both, you achieve the ultimate stage of concurrency. It's like having multiple people, each efficiently switching between tasks, making the most of both parallel execution and idle times. Concurrency is about managing multiple tasks at the same time, whether they're running simultaneously or not."
+> 
+> Excited, John returned to his desk, ready to explore the power of concurrency, understanding how to leverage both parallel and asynchronous programming to make their applications faster and more efficient than ever.
+
+The next twist in this story would likely involve issues with race conditions and language-specific implementation details... But that is not the focus of this article. A better approach is to show a visualization of the definition of concurrency as described in this story:
+
+### The Three Quadrants of Concurrency: A Visual Approach
+
+![](https://cdn.hashnode.com/res/hashnode/image/upload/v1721204052382/54a74571-56c5-4d0d-87fc-5ea8e35ab9eb.png align="center")
+
+This visualization is inspired by [Code Wala](https://codewala.net/2015/07/29/concurrency-vs-multi-threading-vs-asynchronous-programming-explained/). To make it clearer, I added examples from the article into the quadrants. It's a good time to compare it to the visualization by ByteByteGo that I included earlier in this article.
+
+My visualization emphasizes that there are two independent axes in concurrency. But more importantly, it includes async in the definition. This is something rarely seen in definitions or explanations of concurrency. Classics like [Clean Code: A Handbook of Agile Software Craftsmanship](https://learning.oreilly.com/library/view/clean-code-a/9780136083238/), [The Pragmatic Programmer: Your Journey to Mastery](https://learning.oreilly.com/library/view/the-pragmatic-programmer/9780135956977/f_0054.xhtml), and [Concurrent Programming in Java](https://learning.oreilly.com/library/view/concurrent-programming-in/0201310090/pr01.html) (I've checked many other books and papers) do not even mention async. While there were reasons for this omission a decade ago, they are no longer relevant. With the widespread adoption of async/await in most popular languages, ignoring async is no longer justifiable. In modern definitions of concurrency, including async is essential for a complete understanding.
```

Comment:
The problem is that until now there is no one way what is considered concurrency. That is probably the main reason why I wrote this article. My definition aligns quite closely with the definition of Steven Cleary, Brij Bhushan Mishra and many others. I think the problem is that no one really **clearly** stated it.

There are oter popular definitions of concurrency that do use concurrency as a synonym for asynchronous. And that is an other problem, because this contradicts the implementations in Java, C# etc. as mentioned in this article.

I do not want to mention that again here and also I do not want to mention to many alternative definitions as I am worried that this leads to more confusion.

I tried to elaborate about the async part a bit more.